engine/src/distribution/MessageManager.cpp

```diff
@@ -0,0 +1,48 @@
+#include "MessageManager.h"
+#include "utilities/CommonOperations.h"
+
+namespace ral {
+namespace distribution {
+
+int MessageManager::get_total_partition_counts(std::map<std::string, int>& node_count) {
+    int total_count = node_count[node_id];
+    for (auto message : messages_to_wait_for){
+        auto meta_message = input_message_cache->pullCacheData(message);
+        total_count += std::stoi(static_cast<ral::cache::GPUCacheDataMetaData *>(meta_message.get())->getMetadata().get_values()[ral::cache::PARTITION_COUNT]);
+    }
+    return total_count;
+}
+
+void MessageManager::send_total_partition_counts(ral::cache::CacheMachine* graph_output,
```

Comment:
change class to message_manager

Reply:
Most of our class names are in camel case. We don't have one very specific way of naming I think class name should be in camel case and function name should be with underscores.

Reply:
We agreed to follow cudf conventions. The conventions cudf follows are lower case with _ for classes and functions

Reply:
ugh... you are right. we we should just follow cudf. I'm just not very fond of having functions and classes have the same convention

Reply:
I am. and it follows cudf